presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -913,12 +923,12 @@ else if (firstDependencies.stream().allMatch(right.getRelationType().canResolveP
         return output;
     }
 
-    private static boolean isUnnestRelation(Relation node)
+    private static boolean isLateralRelation(Relation node)
     {
         if (node instanceof AliasedRelation) {
-            return isUnnestRelation(((AliasedRelation) node).getRelation());
+            return isLateralRelation(((AliasedRelation) node).getRelation());
         }
-        return node instanceof Unnest;
+        return node instanceof Unnest || node instanceof Lateral;
```

Comment:
No. Lateral relation is something different. Lateral relation it is a relation that has access to fields of left relation. `LATERAL JOIN` or `UNNEST`, both are lateral relation. It is just the name coincidence.